src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1403,6 +1403,41 @@ public async Task DeleteUserAsync(Guid organizationId, Guid userId)
             }
         }
 
+        public async Task DeleteUsersAsync(Guid organizationId, IEnumerable<Guid> organizationUsersId,
+            Guid? deletingUserId)
+        {
+            var orgUsers = await _organizationUserRepository.GetManyAsync(organizationUsersId);
+            var filteredUsers = orgUsers.Where(u => u.OrganizationId == organizationId);
+
+            if (!filteredUsers.Any())
+            {
+                throw new BadRequestException("Users invalid.");
+            }
+
+            var confirmedOwners = (await GetConfirmedOwnersAsync(organizationId)).ToList();
+            var confirmedOwnersIds = confirmedOwners.Select(u => u.Id);
+            if (!confirmedOwnersIds.Except(organizationUsersId).Any())
+            {
+                throw new BadRequestException("Organization must have at least one confirmed owner.");
+            }
```

Comment:
This logic is duplicated in several methods (DeleteUser, SaveUser). Might be worth refactoring out into a private method or something.

Reply:
I've refactored the duplicated code I've found. Haven't fully tested it yet though and we'll want to do some regression testing to verify it behaves the same.

Reply:
Perhaps a nice opportunity to add some test cases for it.